src/common/util/navigator_gpu.ts

```diff
@@ -60,6 +65,64 @@ export function getGPU(recorder: TestCaseRecorder | null): GPU {
 
   impl = gpuProvider();
 
+  if (globalTestConfig.enforceDefaultLimits) {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    const oldFn = impl.requestAdapter;
+    impl.requestAdapter = async function (options?: GPURequestAdapterOptions) {
+      const adapter = await oldFn.call(this, { ...defaultRequestAdapterOptions, ...options });
+      if (adapter) {
+        const limits = Object.fromEntries(
+          Object.entries(getDefaultLimitsForAdapter(adapter)).map(([key, { default: v }]) => [
+            key,
+            v,
+          ])
+        );
+
+        Object.defineProperty(adapter, 'limits', {
+          get() {
+            return limits;
+          },
+        });
+      }
+      return adapter;
+    };
+
+    if (!s_wrappedForEnforceDefaultLimits) {
```

Comment:
Not sure this is necessary? We already have `if (impl) { return impl; }` at the top that should prevent this from running multiple times.

Reply:
You're right, I missed that. I'll remove this check